File: LeetCode/14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
  string str;
  int j = 0;
  if ( strs[j].empty() ) 
    return str;
  char c;
  for ( int i = 0; i < strs.size(); i++ ) {
    if ( i == 0 )
      c = strs[i][j];

    if ( c != strs[i][j] )
      return str;

    if ( i == strs.size() - 1 ) {
      j++;
      i = -1;
      str += c;
      if ( strs[0].length() == j ) 
        return str;
    }
  }
  return str;

    }
};

File: LeetCode/14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
    String str = "";

    int j = 0;
    if ( strs[j].isEmpty() ) 
      return str;
    char c = ' ';
    for ( int i = 0; i < strs.length; i++ ) {
      if ( i == 0 ) 
        c = strs[i].charAt(j);

      if ( strs[i].length() <= j || strs[i].charAt(j) != c ) 
        return str;

      if ( i == strs.length - 1 ) {
        i = -1;
        j++;
        str += c;
        if ( strs[0].length() == j ) 
          return str;
      }
    }
    return str;


    }
}
